weave

```diff
@@ -813,6 +813,16 @@ dns_args() {
     fi
 }
 
+rewrite_hosts() {
+    path=$(docker inspect -f '{{.HostsPath}}' $CONTAINER)
+    name=$(docker inspect -f '{{.Config.Hostname}}' $CONTAINER)
+    containers_path="$(dirname $(dirname "$path"))"
+    docker run --rm \
+        -v "$containers_path":"$containers_path" \
+        --entrypoint=/bin/sh \
+        $EXEC_IMAGE -c "/usr/bin/weavehosts $path $name $ALL_CIDRS"
```

Comment:
Because it depends on the path that leads to this call. `docker run` -> `weaveproxy` -> `weave attach` was ok, because `weaveproxy` is being launched with `-v "$containers_path":"$containers_path"`. But that does not happen when running `weave run` or just `weave attach`: the containers path is not being mounted, so we cannot rewrite `/etc/hosts`